samples/rxfilter/rxfilter.c

```diff
@@ -152,6 +154,36 @@ ParseArgs(
             } else {
                 LOGERR("Unexpected UdpDstPort");
             }
+        } else if (!_stricmp(ArgV[i], "-IcmpDstIpv4")) {
+            if (++i >= ArgC) {
+                LOGERR("Missing IcmpDstIpv4");
+                goto Usage;
+            }
+            if (Rule.Match == XDP_MATCH_ICMPV4_ECHO_REPLY_IP_DST_MASK) {
+                const char* ipStr = ArgV[i];
+                printf("Parsing IP address: %s\n", ipStr);
+
+                // Parse dotted-quad into IN_ADDR (network byte order)
+                int rc = InetPtonA(AF_INET, ipStr, &Rule.Pattern.IpMask.Address.Ipv4);
+                if (rc != 1) {
+                    if (rc == 0) {
+                        printf("Invalid IPv4 address format: %s\n", ipStr);
+                    } else {
+                        printf("InetPtonA error: %d\n", WSAGetLastError());
+                    }
+                    return;
+                }
+                // Sanity check: print back out
+                char buf[INET_ADDRSTRLEN] = {0};
+                if (InetNtopA(AF_INET, &Rule.Pattern.IpMask.Address.Ipv4, buf, (DWORD)sizeof(buf)) == NULL) {
```

Comment:
I don't think we need this sanity test, especially in a sample program. We want this to be as simple and focused as possible.

Reply:
pending fix